adapter/pkg/operator/synchronizer/gateway_synchronizer.go

```diff
@@ -78,10 +88,22 @@ func undeployGateway(gatewayState GatewayState) error {
 func AddOrUpdateGateway(gatewayState GatewayState, state string) (string, error) {
 	gateway := gatewayState.GatewayDefinition
 	resolvedListenerCerts := gatewayState.GatewayStateData.GatewayResolvedListenerCerts
+	gatewayAPIPolicies := gatewayState.GatewayStateData.GatewayAPIPolicies
+	gatewayBackendMapping := gatewayState.GatewayStateData.GatewayBackendMapping
+
+	gwLuaScript, gwReqICluster, gwReqIAddresses, gwResICluster, gwResIAddresses :=
+		generateGlobalInterceptorResource(gatewayAPIPolicies, gatewayBackendMapping)
+
 	if state == constants.Create {
-		xds.GenerateGlobalClusters(gateway.Name)
+		xds.GenerateGlobalClusters(gatewayState.GatewayDefinition.Name)
+	}
+	if state == constants.Create || state == constants.Update {
```

Comment:
could there be any other states in AddOrUpdateGateway method? 
```suggestion
```

Reply:
removed the conditional check